### ISSUE TITLE
Quill-Mentions 

### ISSUE THREAD

**Issue by Kluez** [fgilde/MudExRichTextEditor#1]
hello!

first, i'd like to say that i really like the rich-text-editor which seems like something that mudblazor was missing.

do you have any intention of adding the '[mentions](https://github.com/quill-mention/quill-mention)' module?

best regards
Klaus

**Comment by fgilde** [comment]
Yes good idea. I will add it and also add that you can somehow inject the creation process to register own modules maybe

**Comment by Kluez** [comment]
That sounds great! I hope everything is smooth sailing for the implementation 👍

**Comment by fgilde** [comment]
Ok version version with Mention as optional module is available here in this [Preview package](https://www.nuget.org/packages/MudExRichTextEditor/1.0.4-prev-2)

This then can currently used like this 

```c#
@page "/"
@using MudExRichTextEditor.Extensibility
@using MudExRichTextEditor.Extensibility.Mention
@using Nextended.Core.Extensions
@using BlazorJS
@using Newtonsoft.Json



<MudExRichTextEdit
                   Modules="@_modules"
                   Height="400"
                   EnableResize="true"
                   Class="m-2"
                   @bind-Value="@_value"
                   Placeholder="Write your shit right here">

</MudExRichTextEdit>



@code {

    string _value = "<b>This</b> initial value will change in 2 seconds";
    private IQuillModule[] _modules;
    [Inject] 
    private IJSRuntime _js { get; set; }

    protected override void OnInitialized()
    {
        _modules = new[] {
            new QuillMentionModule<User>((c, s) => Task.FromResult(_users.Where(user => user.FullName.Contains(s, StringComparison.InvariantCultureIgnoreCase))), '@', '$', '#')
                .SetProperties(m => m.MentionClicked = MentionClicked)
        };
        base.OnInitialized();
    }


    private async void MentionClicked(User obj)
    {
        await _js.AlertAsync(JsonConvert.SerializeObject(obj));
    }

    private static IEnumerable<User> _users = new List<User>()
    {
        new() { FirstName = "Florian", LastName = "Gilde", DateOfBirth = new DateTime(1983, 6,8)},
        new() { FirstName = "John", LastName = "Doe", DateOfBirth = new DateTime(1974, 5,11)},
        new() { FirstName = "Jane", LastName = "Doe", DateOfBirth = new DateTime(1977, 12,21) },
        new() { FirstName = "Foo", LastName = "Bar", DateOfBirth = new DateTime(2000, 1,1)  },
        new() { FirstName = "Baz", LastName = "Qux", DateOfBirth = new DateTime(1990, 2,9)  },
    };


    class User
    {
        public int Age => DateTime.Today.Year - DateOfBirth.Year - (DateOfBirth.Date > DateTime.Today.AddYears(-DateOfBirth.Year) ? 1 : 0);
        public DateTime DateOfBirth { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string FullName => $"{FirstName} {LastName}";
        public override string ToString() => FullName;
    }
}

```

But I'am currently not sure if I want to keep using it as additional module and also for paramegters not sure. But you can try and tell me if from the functional side something is missing.

![image](https://github.com/fgilde/MudExRichTextEditor/assets/11070717/ac575f9a-cd8e-4e31-a1ee-5e0210587bca)

![image](https://github.com/fgilde/MudExRichTextEditor/assets/11070717/7b841e9a-f274-4f80-89f3-e069514a1a2b)

**Comment by Kluez** [comment]
I just gave it a quick-test and it is very straight forward to use so far (adding to project, start up and @-test). i will do some more testing / integration and prepare more detailed feedback on the functionality. thanks so far!

**Comment by Kluez** [comment]
First of all, i would like to say again nice and simple it is to integrate. Exchanging for the previously used component was smooth sailing. We think we will use the component (as it is the best so far with mudblazor). We came accross the following issues while using the component.


**QuillMentions** 
When bound to a 'contentField', the mentions element gets a ByteOrderMask added (when hovering over the item)
`const string BOM_MARK = "\ufeff";`
(workaround: we sanitize the content)


**MudExRichTextEdit**

- ReadOnly and Skelleton display

When you have several messages bellow each other (i.e. a comment history) where those entries are readonly,
and you load the CommentHistory component, the skelleton structure is 'too large' so that there is some screen-component-jumping

(workaround: we use other component to display read-only comments)
We would like to be able to adjust (own-renderfragment) or disable it.

- Placeholder
When we used the placeholder, we had an issue in the read-only mode (duplicate text on same position).
Another issue was, when we switched the theme to dark. The placeholder-text does not adjust.
(workaround: no placeholder text)

---
Do you know which way you'll go with the mentions module?
Thanks!

**Comment by fgilde** [comment]
I keep the structure of the modules as it is right now. For the other things you can create new issues (If you are able to try to provide a sample where the errors are easy to reproduce) and I will try to fix. Currently I didnt have so much time for it but I'll try my best. You can also try to fix and contribute here. I released v 1.0.4 today and will close this issue. Please as mentioned create new issues for the placeholder and read only problem you have
